#8 [Feat] MainView 구현

0-jerry opened the pull request:
## #️⃣ Issue Number

#2 

## 📝 요약(Summary)

<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
MainView 의 View와 ViewModel 을 구현했습니다. ( Model 미연동 )


## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)
![Simulator Screen Recording - iPhone 16 Pro - 2025-01-10 at 16 26 41](https://github.com/user-attachments/assets/60d4116b-c09a-497e-b351-5ba1d43dedde)

## 💬 공유사항 to 리뷰어
ViewModel 의 Output 과 Model 을 통해 사용할 형태에 대해서
고려할 점이 있다면 알려주시면 좋을 것 같습니다.
<!--- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이 있으면 적어주세요. -->
<!--- 논의해야할 부분이 있다면 적어주세요.-->
<!--- ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->

## ✅ PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

## ♻️ 추가 변경사항
- [x] Rx 구조개선
- [x] 문자 형식 수정
- [x] 모달 높이 연산 수정

0-jerry committed:
✨[Feat]: WeatherView 구현

0-jerry committed:
🔧 WeatherSimple 모델 디렉토리 분리

0-jerry committed:
Merge branch 'feature/WeatherSimpleView' into feature/mainview

0-jerry committed:
✅ MainView 레이아웃 설정

0-jerry committed:
Merge branch 'develop' into feature/mainView

0-jerry committed:
💄[Style] UI 값 수정 & ✨[Feat] TemperatureFormatter  구현

0-jerry committed:
🔧 폴더 이동 및 에셋 추가

0-jerry committed:
✨[Feat] MainViewModel 구현 및 바인딩

0-jerry committed:
♻️ 뷰모델 리팩토링

0-jerry committed:
✨RefreshDateFormatter 구현

0-jerry committed:
🔀Alamofire 패키지 컨플릭트

soycong commented on GoWalk/GoWalk/MainView/Utility/RefreshDateFormatter.swift:
```diff
@@ -0,0 +1,22 @@
+
+    private static func convert(_ date: Date) -> String {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "HH시 mm분 ss초"
```
작은 디테일까지 신경써주셔서 감사합니다!!!!
요 부분 HH시 mm분까지만 나와도 좋을 것 같습니다!!!! 👍

0-jerry replied:
새로고침이라 조금 더 상세하게 표시해봤는데
데이터를 업데이트한 시점에 대해 초 단위까지 표시할 필요는 없겠네요
감사합니다

soycong reviewed (approved):
여러 디테일 고민하고 제안 후, 적용해주셔서 감사합니다~!!
디테일 모달이랑 연결하면 어떤 모습일지 기대 되네요!! 저도 재영님 코드 참고해서 작성해보아야겠어요!!
고생하셨습니다! 🫡

jeff-0324 commented on GoWalk/GoWalk/MainView/Utility/WeatherAssetTranslator.swift:
```diff
@@ -0,0 +1,19 @@
+//  Created by t2023-m0072 on 1/8/25.
+//
+
+import UIKit.UIImage
```
오? UIImage만 추가하는 방법인가요?

0-jerry replied:
네 특정 클래스에 대해서만 명시할 수 있더라구요

jeff-0324 commented on GoWalk/GoWalk/MainView/View/MainViewController.swift:
```diff
@@ -0,0 +1,235 @@
+    }
+
+    @objc
+    private func footerViewTapped(_ gestureRecognizer: UITapGestureRecognizer) {
```
오 이렇게 원하는 높이만큼 모달을 띄울 수 있군요!

maxminseok commented on GoWalk/GoWalk/MainView/View/MainViewController.swift:
```diff
@@ -0,0 +1,235 @@
+        locationButton.addTarget(self,
+                                 action: #selector(refreshButtonTapped),
+                                 for: .touchUpInside)
+        settingButton.addTarget(self,
```
튜터님 피드백처럼 요기 addTarget부분이랑 밑에 @objc 메서드 Rx로 처리하면 될 거 같은데 어떠신가요..!

0-jerry replied:
넵 수정해서 다시 올려보겠습니다 
감사합니다

maxminseok commented on GoWalk/GoWalk/MainView/View/MainViewController.swift:
```diff
@@ -0,0 +1,235 @@
+                let height = self.view.layer.frame.maxY
+                print(bottomY, height, height - bottomY)
+                // 특정 좌표를 기준으로 잡아야할 필요성이 있음
+                return height - (bottomY + 100) })]
```
이것도 다같이 상의해보면 좋겠네요😁

0-jerry replied:
네 조금 상의해봐야할 것 같아요 기기에 따라서
비율이 달라지는 문제가 있더라구요. 🥲

jeff-0324 reviewed (approved):
상세한 주석과 깔끔하게 코드를 작성하셔서 보기 좋았습니다.
바인딩하신 코드랑 다른 코드들 보면서 많이 배울 수 있는거 같습니다!
구현하시느라 고생하셨습니다!

maxminseok reviewed (approved):
Input/Output 적용하신게 인상깊어요! 많이 배웠습니다! 
고생하셨습니다!!

0-jerry committed:
♻️ Rx 구조 개선

0-jerry committed:
🐝 [Fix]: 상세화면 모달 높이 수정

0-jerry committed:
💡[Comment]: 주석 추가

0-jerry committed:
🎨 [Chore]: 누락된 코드 추가

0-jerry committed:
🔀[Merge]: branch 'develop' of https://github.com/SPT-weather/nbc-weatherApp into feature/mainView

# Conflicts:
#	GoWalk/GoWalk.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
